File: projects/order-creation/core/src/main/java/com/robinfood/core/dtos/transactionrequestdto/PortionDTO.java
package com.robinfood.core.dtos.transactionrequestdto;

import java.io.Serializable;
import java.math.BigDecimal;
import javax.validation.Valid;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import javax.validation.constraints.PositiveOrZero;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PortionDTO implements Serializable {

    private static final long serialVersionUID = 8609026979201840943L;

    @NotNull
    @Min(0)
    private BigDecimal discount;

    @PositiveOrZero
    private Integer free;

    @NotNull
    @Positive
    private Long id;

    @NotNull
    private Boolean isIncluded;

    @NotBlank
    private String name;

    @NotNull
    @Min(0)
    private BigDecimal price;

    @NotNull
    @Valid
    private PortionProductDTO product;

    @PositiveOrZero
    private Integer quantity;

    @Valid
    private ReplacementPortionDTO replacementPortion;

    private String sku;

    @NotNull
    @Positive
    private Long unitId;

    @NotNull
    private Long unitNumber;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/PortionDTO.kt
package com.robinfood.core.dtos.transactionrequest

import java.math.BigDecimal
import javax.validation.Valid
import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Positive
import javax.validation.constraints.PositiveOrZero

data class PortionDTO (

    @NotNull
    @Min(0)
    val discount: BigDecimal,

    @PositiveOrZero
    val free: Int,

    @NotNull
    @Positive
    val id: Long,

    @NotNull
    val isIncluded: Boolean,

    @NotBlank
    val name: String,

    @NotNull
    @Min(0)
    val price: BigDecimal,

    @NotNull
    @Valid
    val product: PortionProductDTO,

    @NotNull
    @Positive
    val quantity: Int,

    @Valid
    val replacementPortion: ReplacementPortionDTO?,

    @NotBlank
    val sku: String,

    @NotNull
    @Positive
    val unitId: Long,

    @NotNull
    @Positive
    val unitNumber: Double
)
